docs/c_sdk_port.md

```diff
@@ -0,0 +1,241 @@
+
+## Application Code for Azure IoT Hub
+
+Let's look at some basic application-level code for using Azure IoT Hub.
+
+### Initialization
+
+#### Azure IoT C SDK
+
+[Taken from the iothub_ll_telemetry_sample](https://github.com/Azure/azure-iot-sdk-c/blob/master/iothub_client/samples/iothub_ll_telemetry_sample/iothub_ll_telemetry_sample.c):
+
+```c
+(void)IoTHub_Init();
+
+IOTHUB_DEVICE_CLIENT_LL_HANDLE device_ll_handle;
+
+// Create the iothub handle here
+device_ll_handle = IoTHubDeviceClient_LL_CreateFromConnectionString(connectionString, MQTT_Protocol);
+
+```
+
+#### Azure IoT middleware for FreeRTOS
+
+[Taken from the sample_azure_iot sample](https://github.com/Azure-Samples/iot-middleware-freertos-samples/blob/main/demos/sample_azure_iot/sample_azure_iot.c):
+
+```c
+ulStatus = prvConnectToServerWithBackoffRetries( ( const char * ) pucIotHubHostname,
+                                                  democonfigIOTHUB_PORT,
+                                                  &xNetworkCredentials, &xNetworkContext );
+/* Fill in Transport Interface send and receive function pointers. */
+xTransport.pxNetworkContext = &xNetworkContext;
+xTransport.xSend = TLS_Socket_Send;
+xTransport.xRecv = TLS_Socket_Recv;
+
+xResult = AzureIoTHubClient_Init( &xAzureIoTHubClient,
+                                  pucIotHubHostname, pulIothubHostnameLength,
+                                  pucIotHubDeviceId, pulIothubDeviceIdLength,
+                                  &xHubOptions,
+                                  ucMQTTMessageBuffer, sizeof( ucMQTTMessageBuffer ),
+                                  ullGetUnixTime,
+                                  &xTransport );
+```
+
+Notice in the case of the Azure IoT C SDK, the full stack of networking is taken care of by Microsoft, so you only need to initialize the client with a connection string containing your credentials along with the protocol of your choice (MQTT, AMQP, or HTTP). After that point, you are ready to send telemetry messages or use any other Azure IoT features.
+
+With the Azure IoT middleware for FreeRTOS, the initialization of the client application requires the user to pass a network context which the MQTT layer uses to send and receive MQTT messages. This means that the network context must be initialized and connected by the user before passing it to the middleware (in this case a static function called `prvConnectToServerWithBackoffRetries()` handles the initialization of that context). To see what that initialization entails, please see [the link here to that static function](https://github.com/Azure-Samples/iot-middleware-freertos-samples/blob/bea90b2a7ce11dc06612bdce142b90f59f436be1/demos/sample_azure_iot/sample_azure_iot.c#L558-L612).
```

Comment:
We'll still have the problem of that function having its name changed in that case. I think it's sort of a pick your poison situation so here I'm inclined to have it directly link to the implementation since at that point it will have a useful code snippet. Eventually, if this is stale, it'll get update if someone finds issue with it.